### PR TITLE
Feature/autowiring release 3

### PR DESCRIPTION
**Pull request by iruzevic**
I have:

- [ ] fixed linting issues
- [ ] Implemented fix to exclude view files and non php files
- [ ] fixed some issues with missing names and etc.

This is tested on actual project with BP 5.0.0 and it works

### CODE REVIEW

File: src/Main/AbstractMain.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ public function build_di_container() {
    * @return array<array>
    */
   private function get_service_classes_with_autowire() {
-    return array_merge( $this->autowiring->buildServiceClasses(), $this->get_service_classes() );
+    return array_merge( $this->autowiring->build_service_classes(), $this->get_service_classes() );
```
Comment (dingo-d) [inline src/Main/AbstractMain.php:103]: PSR-4 it should be `buildServiceClasses()` 🤔

**Review by dingo-d — COMMENTED** [review]
Why did you replace method names to snake_case instead of camelCase?

File: src/Main/AbstractMain.php (hunk 1 above)
Comment (iruzevic) [reply src/Main/AbstractMain.php]: fixed

**Comment by iruzevic** [comment]
> Why did you replace method names to snake_case instead of camelCase?

Didn't knew about the new psr-12 standard. It is updated

File: src/Main/Autowiring.php [hunk 2]
```diff
@@ -1,335 +1,337 @@
...
+		if (! is_dir($pathToNamespace)) {
+			return [];
+		}
+
+		$it = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($pathToNamespace));
+		foreach ($it as $file) {
+			if ($file->isDir()) {
+				continue;
+			}
+			if (preg_match('/\.(php)$/', $file->getFileName()) && preg_match('/^[A-Z]{1,}.*$/m', $file->getFileName())) {
```
Comment (IvanGrginovInf) [inline src/Main/Autowiring.php:192]: I think you should be able to do it as a single regex experssion like so (all filename starting with capital letter, ending with `.php`)

```suggestion
			if (preg_match('/[A-Z]{1}.*.php/', $file->getFileName())) {
```

**Review by IvanGrginovInf — CHANGES_REQUESTED** [review]
Left a single comment regarding simplifying regex, otherwise looks good.

File: src/Main/Autowiring.php (hunk 2 above)
Comment (iruzevic) [reply src/Main/Autowiring.php]: Tnx I have implemented the fix

**Review by IvanGrginovInf — APPROVED** [review]

**Comment by IvanGrginovInf** [comment]
@iruzevic Approved but please fix the conflict before mering.

**Pull request merged**